gym-http-api build failure with latest servant

As seen on the Stackage build server.

Ping Sam Stites <sam@stites.io> @stites. I didn't see an issue tracker to report this on, so reporting here.

```
Building executable 'example' for gym-http-api-0.1.0.0..
[1 of 1] Compiling Main             ( examples/Agent.hs, dist/build/example/example-tmp/Main.o )

examples/Agent.hs:26:30: error:
    • Couldn't match expected type ‘ClientEnv’
                  with actual type ‘Maybe (GHC.Conc.Sync.TVar CookieJar)
                                    -> ClientEnv’
    • Probable cause: ‘ClientEnv’ is applied to too few arguments
      In the second argument of ‘runClientM’, namely
        ‘(ClientEnv manager url)’
      In a stmt of a 'do' block:
        out <- runClientM example (ClientEnv manager url)
      In the expression:
        do manager <- newManager defaultManagerSettings
           out <- runClientM example (ClientEnv manager url)
           case out of
             Left err -> print err
             Right ok -> print ok
   |
26 |   out <- runClientM example (ClientEnv manager url)
   |                              ^^^^^^^^^^^^^^^^^^^^^
```

--- a/build-constraints.yaml
+++ b/build-constraints.yaml
@@ -3256,8 +3256,7 @@ packages:
         - hedgehog-corpus
 
     "Sam Stites <sam@stites.io> @stites":
-        []
-        # - gym-http-api # https://github.com/fpco/stackage/issues/3354
+        - gym-http-api
 
     "Tom Sydney Kerckhove <syd.kerckhove@gmail.com> @NorfairKing":
         - genvalidity

